[HIPIFY][7.2][RT] Sync with `ROCm HIP 7.2` - Part 1 - HIP Runtime

ranapratap55 opened the pull request:
- Added new `hipDeviceAttributeHostNumaId, hipStreamCopyAttributes(), hipLibraryEnumerateKernels(), hipKernelGetLibrary(), hipKernelGetName(), hipOccupancyAvailableDynamicSMemPerBlock()`
- Updated synthetic tests, regenerated `hipify-perl, Driver and Function CUDA2HIP` docs accordingly

ranapratap55 committed:
[HIPIFY][RT][7.2.0] Sync with HIP 7.2.0 Driver and Functions

ranapratap55 committed:
[HIPIFY][#2169][cmake][cuDNN][cuTENSOR]Fix CMake find_file() with NO_DEFAULT_PATH and cache clearing

emankov commented on src/CUDA2HIP_Driver_API_functions.cpp:
```diff
@@ -198,13 +198,13 @@ const std::map<llvm::StringRef, hipCounter> CUDA_DRIVER_FUNCTION_MAP {
   {"cuKernelSetAttribute",                                        {"hipKernelSetAttribute",                                       "", CONV_LIBRARY, API_DRIVER, SEC::LIBRARY, HIP_UNSUPPORTED}},
   {"cuKernelSetCacheConfig",                                      {"hipKernelSetCacheConfig",                                     "", CONV_LIBRARY, API_DRIVER, SEC::LIBRARY, HIP_UNSUPPORTED}},
-  {"cuKernelGetName",                                             {"hipKernelGetName",                                            "", CONV_LIBRARY, API_DRIVER, SEC::LIBRARY, HIP_UNSUPPORTED}},
+  {"cuKernelGetName",                                             {"hipKernelGetName",                                            "", CONV_LIBRARY, API_DRIVER, SEC::LIBRARY}},
```
Could you please also mark all the newly appeared HIP Runtime APIs (functions and datatypes) as `HIP_EXPERIMENTAL`?
Additionally, you'll need to switch the ` HIP_LATEST` to the new `HIP_7020` and add the versioning for each new HIP RT API, for instance:
```cpp
  {"hipKernelGetName",  {HIP_7020, HIP_0, HIP_0, HIP_LATEST}},
```

ranapratap55 replied:
added and updated.

emankov commented on docs/reference/tables/CUDA_Driver_API_functions_supported_by_HIP.md:
```diff
@@ -456,7 +456,7 @@
 |`CU_DEVICE_ATTRIBUTE_HOST_MEMORY_POOLS_SUPPORTED`|13.0| | | | | | | | | | |
 |`CU_DEVICE_ATTRIBUTE_HOST_NATIVE_ATOMIC_SUPPORTED`|8.0| | | |`hipDeviceAttributeHostNativeAtomicSupported`|4.3.0| | | | | |
-|`CU_DEVICE_ATTRIBUTE_HOST_NUMA_ID`|12.2| | | | | | | | | | |
+|`CU_DEVICE_ATTRIBUTE_HOST_NUMA_ID`|12.2| | | |`hipDeviceAttributeHostNumaId`|7.2.0| | | | | |
```
Have the `md` docs been generated by `hipify-clang`?
The question is prompted by the lack of code in the PR to set HIP API versions explicitly.
For instance:
```cpp
{"hipDeviceAttributeHostNumaId",  {HIP_7020, HIP_0, HIP_0, HIP_LATEST}},
```

ranapratap55 replied:
added HIP_7020 and regenerated docs.

emankov committed:
[HIPIFY][doc] `LLVM 21.1.6` is the latest stable supported versions

+ No patches are needed
+ Updated the `README.md` accordingly
+ Tested on `Windows 11` (latest `VS 2019` and `VS 2022`)  and `Ubuntu 24.04.2` (`GNU C/C++ 13.3`)

ranapratap55 committed:
Added HIP_EXPERIMENTAL, HIP_7020 to CUDA2HIP_Driver/Functions and regenerated hipify-perl, CUDA2HIP_Driver/Function docs

ranapratap55 committed:
Added HIP_EXPERIMENTAL, HIP_7020 to CUDA2HIP_Driver/Functions and regenerated hipify-perl, CUDA2HIP_Driver/Function docs

emankov commented on src/Statistics.h:
```diff
@@ -420,7 +420,8 @@ enum hipVersions {
   HIP_7010 = 7010,
-  HIP_LATEST = HIP_7010,
+  HIP_7020 = 7020,
+  HIP_LATEST = HIP_7020,
```
Let's keep `HIP_LATEST` as `7010` until we finish with it, so we don't mix `7010` and `7020` APIs while finishing their support. It will need to regenerate pods and hipify-perl. Thank you very much!

ranapratap55 replied:
updated and moved to new PR #2191 as this one is having merge issues.

emankov committed:
[HIPIFY][7.2][doc] `CHANGELOG.md` update

dependabot[bot] committed:
Bump rocm-docs-core from 1.29.0 to 1.30.0 in /docs/sphinx

Bumps [rocm-docs-core](https://github.com/ROCm/rocm-docs-core) from 1.29.0 to 1.30.0.
- [Release notes](https://github.com/ROCm/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/ROCm/rocm-docs-core/compare/v1.29.0...v1.30.0)

---
updated-dependencies:
- dependency-name: rocm-docs-core
  dependency-version: 1.30.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

emankov committed:
[HIPIFY][7.1][FFTw] Sync with `ROCm HIP 7.1` - Part 2 - `fftw` - Step 6 - Functions - final

+ Updated synthetic tests, the regenerated `hipify-perl`, and `FFT` `CUDA2HIP` docs accordingly

ranapratap55 committed:
Merge branch 'amd-develop' into users/ranapratap55/cuda2hip_runtime_api_7-2-0
